src/extensions/rich-text/code.ts

```diff
@@ -0,0 +1,10 @@
+import { Code } from '@tiptap/extension-code'
+
+const RichTextCode = Code.extend({
+    // Instead of not allowing other marks to coexist, allow everything so that
+    // inline code can be bolded, linked, etc
+    // See https://tiptap.dev/api/schema#excludes
```

Comment:
My initial thought was to have `excludes: 'bold italic strike'`, basically only allowing the link mark (which was what the initial issue, and our internal issue, was about). However, this will provide us with a behavior more close to GitHub's, which is a good thing (from my point of view 😅).

Anyway, for documentation and consistency purposes, I propose the following JSDoc:

```suggestion
/**
 * Custom extension that extends the built-in `Code` extension to allow all marks (e.g., Bold,
 * Italic, and Strikethrough) to coexist with the `Code` mark (as opposed to disallowing all any
 * other mark by default).
 *
 * @see https://tiptap.dev/api/schema#excludes
 */
const RichTextCode = Code.extend({
```

Reply:
Also, it's probably worth checking if our Markdown styles (I believe they now live on Reactist, under the `<Prose>` component) have proper support to style inline code marks with surrounding bold, italic and strikethrough marks.

Reply:
We may also need to apply the highlight colour correctly to the hover styles, but the idle state looks correct:

![image](https://github.com/Doist/typist/assets/8531248/9714b07a-dbd2-42cc-b7e0-f828e2fa0daa)

![image](https://github.com/Doist/typist/assets/8531248/a2e44f15-7248-4b81-a840-9c2c707a4219)